fuel/transformers/__init__.py

```diff
@@ -420,7 +420,22 @@ def get_epoch_iterator(self, **kwargs):
         batches = chain.from_iterable(
             izip(*[data_stream.get_epoch_iterator()
                    for data_stream in self.data_streams]))
-        return partition(len(self.sources), chain.from_iterable(batches))
+
+        part = partition(len(self.sources), chain.from_iterable(batches))
+        as_dict = kwargs.get('as_dict', False)
+        if as_dict:
+            return imap(make_dict(self.sources), part)
```

Comment:
Does something like `imap(dict, imap(zip, izip(part, repeat(self.sources)))` work? Is a bit more straightforward than having a separate class.